File: ruby/selection_sort.rb
# require 'pry'

def selection_sort(arr)
  ans = []
  arr.length.times do
    lowest = arr[0]
  arr.each do |num|
    if num < lowest
      lowest = num
      end
    end
    ans << lowest
    arr.delete(lowest)
  end
  ans
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: [-1, 2, 3, 5]"
  print "=> "
  print selection_sort([3, -1, 5, 2])

  puts

  # Don't forget to add your own!

  # BENCHMARK HERE, you can print the average runtime
  long_input = []

  100.times { long_input << rand }
end

Please add your pseudocode to this file
And a written explanation of your solution

File: javascript/selection_sort.js
function selectionSort(arr) {
  let startArray = [...arr]
  console.log(`startArray: ${startArray}`)
  let newArray = [];
  for (let i = 0; i < arr.length; i++) {
    const lowest = Math.min(...startArray)
    console.log(`lowest: ${lowest}`)
    newArray.push(lowest)
    console.log(`newArray: ${newArray}`)
    lowestIndex = startArray.indexOf(lowest)
    console.log(`lowestIndex: ${lowestIndex}`)
    startArray.splice(lowestIndex, lowestIndex)
    console.log(`startArray after splice ${startArray}`)
console.log(startArray)
    

  }
  return newArray;
}

if (require.main === module) {
  // add your own tests in here
  console.log('Expecting: [-1, 2, 3, 5]');
  console.log('=>', 

  console.log('');

  // BENCHMARK HERE, and print the average runtime
  const longInput = [];

  for (let i = 0; i < 100; ++i) {
    longInput.push(Math.random());
  }
}

module.exports = selectionSort;

// Please add your pseudocode to this file

// let new array = []
// for i in array length:
// push math.min(array.slice(i)) to array

// And a written explanation of your solution
